File: backend/controllers/PaymentController.js
const Payment = require("../models/Payment");

// Get All Payments
const getAllPayments = async (req, res, next) => {
    try {
        const payments = await Payment.getAllPayments();

        res.status(200).json({
            success: true,
            count: payments.length,
            data: payments,
        });
    } catch (error) {
        next(error);
    }
};

module.exports = {
    getAllPayments,
};

File: backend/src/main/java/com/procureease/backend/controller/PaymentController.java
package com.procureease.backend.controller;

import com.procureease.backend.dto.PaymentResponse;
import com.procureease.backend.service.PaymentService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/api/payments")
@RequiredArgsConstructor
public class PaymentController {

    private final PaymentService paymentService;

    // ============================================================
    // Get All Payments
    // ============================================================

    @GetMapping
    public ResponseEntity<Map<String, Object>> getAllPayments() {

        List<PaymentResponse> payments =
                paymentService.getAllPayments();

        Map<String, Object> response =
                new HashMap<>();

        response.put("success", true);
        response.put("count", payments.size());
        response.put("data", payments);

        return ResponseEntity.ok(response);
    }
}